src/main/php/web/io/WriteChunks.class.php

```diff
@@ -44,9 +44,19 @@ public function write($chunk) {
     }
   }
 
+  /** @return void */
+  public function flush() {
+    if (strlen($this->buffer) > 0) {
+      $this->target->write(dechex(strlen($this->buffer))."\r\n".$this->buffer."\r\n");
+      $this->buffer= '';
+      $this->target->flush();
+    }
+  }
+
   /** @return void */
   public function finish() {
-    $this->target->write(dechex(strlen($this->buffer))."\r\n".$this->buffer."\r\n0\r\n\r\n");
```

Comment:
This was broken if the buffer contained no bytes, yielding *two* zero-length chunks (`0\r\n\r\n`) at the end and potentially causing https://github.com/xp-forge/web/issues/18